File: matiasmasso/Mat.NET/DTO/DTOTransportista.cs
using System;

namespace DTO
{
    public class DTOTransportista : DTOContact
    {
        public string Abr { get; set; }
        public int Cubicaje { get; set; }
        public string TrackingUrlTemplate { get; set; }
        public bool Activat { get; set; }

        public decimal noCubicarPerSotaDe { get; set; }
        public int compensaPercent { get; set; }
        public bool allowReembolsos { get; set; }
        public bool transportaMobiliari { get; set; }

        public enum Wellknowns
        {
            souto,
            tnt,
            seur,
            txt
        }

        public enum CodTarifa
        {
            preuHastaKg,
            preuPerKg
        }

        public DTOTransportista() : base()
        {
        }

        public DTOTransportista(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOTransportista Wellknown(Wellknowns id)
        {
            DTOTransportista retval = null;
            string sGuid = "";
            switch (id)
            {
                case Wellknowns.souto:
                    {
                        sGuid = "4D92D321-C045-49D6-BEAD-181169002128";
                        break;
                    }

                case Wellknowns.tnt:
                    {
                        sGuid = "9EE52C83-BFCE-41C2-9435-8D0516C26424";
                        break;
                    }
                case Wellknowns.txt:
                    {
                        sGuid = "8FC74DDC-C659-4EEE-9AFA-D79A6900E5A8";
                        break;
                    }
            }

            if (sGuid.isNotEmpty())
            {
                Guid oGuid = new Guid(sGuid);
                retval = new DTOTransportista(oGuid);
            }
            return retval;
        }


        public static DTOTransportista FromContact(DTOContact oContact)
        {
            DTOTransportista retval = null;
            if (oContact != null)
            {
                retval = new DTOTransportista(oContact.Guid);
                {
                    var withBlock = retval;
                    withBlock.Nom = oContact.Nom;
                    withBlock.NomComercial = oContact.NomComercial;
                    withBlock.FullNom = oContact.FullNom;
                    withBlock.Nifs = oContact.Nifs;
                    withBlock.Address = oContact.Address;
                    withBlock.ContactClass = oContact.ContactClass;
                    withBlock.Lang = oContact.Lang;
                    withBlock.Rol = oContact.Rol;
                }
            }
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOTransportista.vb
Public Class DTOTransportista
    Inherits DTOContact

    Property Abr As String
    Property Cubicaje As Integer
    Property NoCubicarPerSotaDe As Decimal
    Property CompensaPercent As Integer
    Property AllowReembolsos As Boolean
    Property TransportaMobiliari As Boolean
    Property Activat As Boolean

    Public Enum wellknowns
        Souto
        Tnt
        Seur
    End Enum

    Public Enum CodTarifa
        PreuHastaKg
        PreuPerKg
    End Enum

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function wellknown(id As wellknowns) As DTOTransportista
        Dim retval As DTOTransportista = Nothing
        Dim sGuid As String = ""
        Select Case id
            Case wellknowns.Souto
                sGuid = "4D92D321-C045-49D6-BEAD-181169002128"
            Case wellknowns.Tnt
                sGuid = "9EE52C83-BFCE-41C2-9435-8D0516C26424"
        End Select

        If sGuid > "" Then
            Dim oGuid As New Guid(sGuid)
            retval = New DTOTransportista(oGuid)
        End If
        Return retval
    End Function

    Shared Function FromContact(oContact As DTOContact) As DTOTransportista

        Dim retval As DTOTransportista = Nothing
        If oContact Is Nothing Then
            retval = New DTOTransportista
        Else
            retval = New DTOTransportista(oContact.Guid)
            With retval
                .Nom = oContact.Nom
                .NomComercial = oContact.NomComercial
                .FullNom = oContact.FullNom
                .Nif = oContact.Nif
                .Address = oContact.Address
                .ContactClass = oContact.ContactClass
                .Lang = oContact.Lang
                .Rol = oContact.Rol
            End With
        End If
        Return retval
    End Function
End Class
